src/m365/planner/commands/bucket/bucket-remove.ts

```diff
@@ -0,0 +1,255 @@
+import { Cli, Logger } from '../../../../cli';
+import {
+  CommandOption
+} from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import Utils from '../../../../Utils';
+import { GraphItemsListCommand } from '../../../base/GraphItemsListCommand';
+import { Bucket } from '../../Bucket';
+import commands from '../../commands';
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  bucketId: string;
+  bucketName: string;
+  planId?: string;
+  planName?: string;
+  ownerGroupId?: string;
+  ownerGroupName?: string;
+  confirm?: boolean;
+}
+
+class PlannerBucketRemoveCommand extends GraphItemsListCommand<any> {
+  public get name(): string {
+    return commands.BUCKET_REMOVE;
+  }
+
+  public get description(): string {
+    return 'Removes the Microsoft Planner bucket from a plan';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.bucketId = typeof args.options.bucketId !== 'undefined';
+    telemetryProps.bucketName = typeof args.options.bucketName !== 'undefined';
+    telemetryProps.planId = typeof args.options.planId !== 'undefined';
+    telemetryProps.planName = typeof args.options.planName !== 'undefined';
+    telemetryProps.ownerGroupId = typeof args.options.ownerGroupId !== 'undefined';
+    telemetryProps.ownerGroupName = typeof args.options.ownerGroupName !== 'undefined';
+    telemetryProps.confirm = (!(!args.options.confirm)).toString();
+    return telemetryProps;
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['bucketId', 'bucketName', 'planId', 'confirm'];
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: (err?: any) => void): void {
+    const removeBucket: () => void = (): void => {
+      if (args.options.bucketName) {
```

Comment:
Let's try to avoid code duplication for removing the bucket. We can do this by moving all code responsible for returning the bucket into the `getBucket` function. In there, we can have two code paths: return the bucket by ID and by name. In the end, we get a single bucket (or an error for bucket not found) which can return here for deletion and which we could do with a single call.